routers/web/repo/view.go

```diff
@@ -728,16 +728,65 @@ func checkHomeCodeViewable(ctx *context.Context) {
 	ctx.NotFound("Home", fmt.Errorf(ctx.Tr("units.error.no_unit_allowed_repo")))
 }
 
+func checkCitationFile(ctx *context.Context) {
+	// Check if repo is not empty and if we are on repo src
+	if ctx.Repo.Repository.IsEmpty {
+		return
+	}
+	entry, err := ctx.Repo.Commit.GetTreeEntryByPath(ctx.Repo.TreePath)
+	if err != nil {
+		ctx.NotFoundOrServerError("Repo.Commit.GetTreeEntryByPath", git.IsErrNotExist, err)
+		return
+	}
+	if entry.Name() != "" {
+		return
+	}
+	tree, err := ctx.Repo.Commit.SubTree(ctx.Repo.TreePath)
+	if err != nil {
+		ctx.NotFoundOrServerError("Repo.Commit.SubTree", git.IsErrNotExist, err)
+		return
+	}
```

Comment:
```suggestion
func checkCitationFile(ctx *context.Context) {
	// Check if repo is not empty and if we are on repo src
	if ctx.Repo.Repository.IsEmpty {
		return
	}
	tree, err := ctx.Repo.Commit.SubTree(ctx.Repo.TreePath)
	if err != nil {
		ctx.NotFoundOrServerError("Repo.Commit.SubTree", git.IsErrNotExist, err)
		return
	}
```

This is what I meant, as this code would likely work(given use it like this at other places in Gitea).

Reply:
> This is what I meant, as this code would likely work(given use it like this at other places in Gitea).

I understand your suggestions better, but `entry, err := ctx.Repo.Commit.GetTreeEntryByPath(ctx.Repo.TreePath)` allows me to know if we are on the root of the project with this line `if entry.Name() != "" ` and if we are not, we do not need to continue the execution of the function because my feature is only available on the repo's home page.

Reply:
Ah, okay I see. Thank you for your explanation. So in order to avoid calling that function we can do a simple check and combine it at line 733:

```go
if ctx.Repo.Repository.IsEmpty || ctx.Repo.TreePath == "" {
```

Reply:
Ah okay my bad, I just tested a bit around and seems like the check is still necessary(but for other reason) as it would otherwise 404 when viewing a file. Apologies.

Reply:
> Ah okay my bad, I just tested a bit around and seems like the check is still necessary(but for other reason) as it would otherwise 404 when viewing a file. Apologies.

Perfect, it's okay, thanks for you time. Do you know if this PR will be merge, if yes when ?

Reply:
>  Do you know if this PR will be merge, if yes when ?

It requires at least two approvals from different maintainers. So far nobody else has been active on this PR so it might take a while.